### DISCUSSION TITLE
Monitor disk usage per user?

### DISCUSSION THREAD

**Discussion by zemmyang** [netdata/netdata#15156] [Q&A]
What I do right now is to run `du --max-depth=1 -h` on some root directory, then `ls -l` to get the user who owns those directories, then form that to a chart.

Does netdata have a way do automate the creation of a chart with that kind of information? It doesn't have to be real-time - I check once every week, for example. 

I can see that I can write that using the pandas data collection option, but maybe there's something already built-in?

(We really want to know who's hogging space in our network drives)

**Comment by thiagoftsm** [comment]
Hello @zemmyang,

Our disk plugin is not collecting nothing in this level yet, but there are possibilities to monitor this following one of next steps:

- It is possible to create an [external plugin](https://learn.netdata.cloud/docs/contributing-to-netdata/external-plugins/) that runs the command you gave us and parse the output. 
- We could create  a plugin to monitor quotas per user and per group, but in this case users must configure quota per partition.

The suggestions about  could be written in any programming language.

 I am pinging (@shyamvalsan , @sashwathn ) our product team for we proceed with discussion.

Best regards!

**Reply by sashwathn** [reply]
@thiagoftsm Are we not collecting disks related information in our users section where we have cpu, memory and network related contexts per user?

**Reply by thiagoftsm** [reply]
Running default, I do not think so, this is the motive I said we could monitor quotas. 
I cannot see this information on my hosts at least, do you have in yours?

**Reply by Ferroin** [reply]
@sashwathn We _can’t_ collect disk space usage for that, because it’s completely unrelated to how we’re collecting the rest of the data. There is no cgroup for tracking disk space usage.

**Reply by Ferroin** [reply]
As far as actually collecting this, the correct solution is indeed filesystem quotas.

_However_... That’s not actually as simple as it sounds, because at least BTRFS and ZFS have _drastically_ different quota systems from the classic ext4 setup, and even if we ignore those we have to figure out what we want to do with project quotas (which are only supported on some filesystems).

**Comment by andrewm4894** [comment]
I think this could be relevant and useful as a feature in functions in some way. Will cross link that discussion.

https://github.com/netdata/netdata/discussions/14412

**Reply by thiagoftsm** [reply]
I agree. Nowadays we have "giant" emails, but in ancient time, I remember when I had to configure quota of 5Mb per user. 
There are many other cases where quota is necessary.
